### PR TITLE
fix(api gateway lambda integration): Fix redeployment trigger

### PR DESCRIPTION
**Pull request by kkyeboah**
## Goal

- JSON encode the `PocketApiGatewayProps` to trigger redeployment of the API Gateway resource
- Add tags to the API Gateway REST API resource

## Implementation decisions
- redeployment triggers are provided by the service consuming the API gateway construct. if no trigger is provided, then we use the current time as the trigger. this way, what's being set as a trigger becomes the responsibility of the consuming service

ticket:[ infra-188 ](https://getpocket.atlassian.net/browse/INFRA-188)

### CODE REVIEW

File: src/pocket/PocketApiGatewayLambdaIntegration.ts [hunk 1]
```diff
@@ -55,6 +55,15 @@ export class PocketApiGateway extends Resource {
...
       route.lambda.lambda.versionedLambda,
     ]);
+
+    const apiGatewayRoutes = config.routes;
+    const apiGatewayRoutesObject = {};
+    apiGatewayRoutes.forEach(
+      (route, index) => (apiGatewayRoutesObject[index] = route)
+    );
+    //remove the routes property from the config object
+    delete config.routes;
```
Comment (bkochendorfer) [inline src/pocket/PocketApiGatewayLambdaIntegration.ts:65]: ahh

File: src/pocket/PocketApiGatewayLambdaIntegration.ts [hunk 2]
```diff
@@ -69,7 +78,10 @@ export class PocketApiGateway extends Resource {
           redeployment: Fn.sha1(
             Fn.jsonencode({
               resources: routeDependencies.map((d) => d.id),
-              config: JSON.stringify(config),
+              config: JSON.stringify({
```
Comment (bkochendorfer) [inline src/pocket/PocketApiGatewayLambdaIntegration.ts:81]: I hate how tricky this is in general.

**Review by bkochendorfer — DISMISSED** [review]

File: src/pocket/PocketApiGatewayLambdaIntegration.ts (hunk 2 above)
Comment (bkochendorfer) [reply src/pocket/PocketApiGatewayLambdaIntegration.ts:81]: Not your code, AWS API Gateway behavior

Comment (sri-kirsh) [reply src/pocket/PocketApiGatewayLambdaIntegration.ts:81]: ya we have to recursively make sure there are no list in the `config` object.

**Review by sri-kirsh — APPROVED** [review]
🍐

File: src/pocket/PocketApiGatewayLambdaIntegration.ts [hunk 3]
```diff
@@ -53,6 +61,11 @@ export class PocketApiGateway extends Resource {
       route.resource,
       route.lambda.lambda.versionedLambda,
     ]);
+
+    // Use the current timestamp in milliseconds to trigger a redeployment if no triggers are provided.
+    // If no triggers are provided, this resource will be redeployed on every "terraform apply".
+    const triggers = config.triggers ?? { deployedAt: Date.now().toString() };
```
Comment (kschelonka) [inline src/pocket/PocketApiGatewayLambdaIntegration.ts:67]: works for me

**Review by kschelonka — APPROVED** [review]
annoying, but we probably want to make a new deployment every time anyways... so pretty slick solution there

**Pull request merged**